ftw/referencewidget/resources/refwidget.js

```diff
@@ -0,0 +1,178 @@
+$(function() {
```

Comment:
You don't have to wrap the whole code in a jQuery-Ready event.
If you use event-delegation like:
```javascript
$(document).on("eventtype", "target", func);
```

Because the document is always available in browser scope it doesn't matter when the element appears in the DOM

You just have to wrap the check for the presence of the reference widget in the jQuery-Ready event:
```javascript
$(function() {
  if($('.referencewidget').length) { return false; }
});
```